registry/storage/signedmanifesthandler.go

```diff
@@ -36,6 +43,14 @@ func (ms *signedManifestHandler) Unmarshal(ctx context.Context, dgst digest.Dige
 		return nil, err
 	}
 
+	if ms.repository.schema1SigningKey != nil {
+		if err := jsig.Sign(ms.repository.schema1SigningKey); err != nil {
+			return nil, err
+		}
+	} else if !ms.repository.schema1SignaturesEnabled {
+		return nil, fmt.Errorf("Missing signing key with signature store disabled")
```

Comment:
Oh, I did a grep for `'fmt.Errorf("[A-Z]'` and found quite a few examples. They are mostly in `registry/storage/driver` though.